src/wallet/wallet.cpp

```diff
@@ -3434,7 +3440,7 @@ bool CWallet::ReserveKeyFromKeyPool(int64_t& nIndex, CKeyPool& keypool, bool fRe
         if (!batch.ReadPool(nIndex, keypool)) {
             throw std::runtime_error(std::string(__func__) + ": read failed");
         }
-        if (!HaveKey(keypool.vchPubKey.GetID())) {
+        if (!IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS) && !HaveKey(keypool.vchPubKey.GetID())) {
```

Comment:
Belt and suspenders, we should be checking that this exists in our wallet regardless yes? Just not asking if we have the private key itself.

Reply:
I changed this line to check for the pubkey in the wallet instead of the privkey.